packages/core/src/state-management/state-manager.ts

```diff
@@ -323,12 +324,16 @@ export class StateManager {
       return
     }
 
-    await this._saveAnchorRequestForState(state$)
-    const anchorStatus$ = this.anchorService.requestAnchor({
-      streamID: state$.id,
-      tip: state$.tip,
-      timestampISO: (new Date()).toISOString(),
-    })
+    const genesisCID = state$.value.log[0].cid
+    const genesisCommit = await this.dispatcher.retrieveCommit(genesisCID, state$.id)
+    await this._saveAnchorRequestForState(state$, genesisCommit)
+
+    const requestAnchorParams = await this._buildRequestAnchorParams(
+      state$,
+      genesisCID,
+      genesisCommit
+    )
+    const anchorStatus$ = this.anchorService.requestAnchor(requestAnchorParams)
```

Comment:
This code could use some clean up. I would suggest something like this:
```suggestion
    const requestCar = await this._buildAnchorRequestCarFromState(state$)
    
    const genesisCID = state$.value.log[0].cid
    const genesisCommit = requestCar.get(genesisCID)
    await this._saveAnchorRequestForState(state$, genesisCommit)

    const anchorStatus$ = this.anchorService.requestAnchor(requestCar)
```

This removes the need for the `RequestAnchorParams`. I don't believe gathering the params and creating the CAR file needs to be two different steps, in two different files. Hopefully this should make the code more readable!